src/App.jsx

```diff
@@ -1,28 +1,41 @@
 import { Outlet } from 'react-router'
 import { Routes, Route, BrowserRouter } from 'react-router-dom'
-import CssBaseline from '@mui/material/CssBaseline';
+import CssBaseline from '@mui/material/CssBaseline'
 import { ThemeProvider } from '@mui/material'
 import { createTheme } from '@mui/material/styles'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 import MainPage from './pages/MainPage'
 import DetailPage from './pages/DetailPage'
-import SearchPage from './pages/SearchPage'
-import SearchDetailPage from './pages/SearchDetailPage'
+import SearchPage from './pages/search/SearchPage'
+import SearchDetailPage from './pages/search/SearchDetailPage'
 
 function App() {
   const theme = createTheme()
 
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        staleTime: Infinity,
+        cacheTime: Infinity,
+      },
+    },
+  })
+
   return (
     <ThemeProvider theme={theme}>
       <CssBaseline>
         <BrowserRouter>
-          <Routes>
-            <Route element={<Outlet />}>
-              <Route path="/" element={<MainPage />} />
-              <Route path="/detail" element={<DetailPage />} />
-              <Route path="/search" element={<SearchPage />} />
-              <Route path="/search-detail" element={<SearchDetailPage />} />
-            </Route>
-          </Routes>
+          <QueryClientProvider client={queryClient}>
+            <Routes>
+              <Route element={<Outlet />}>
+                <Route path="/" element={<MainPage />} />
+                <Route path="/detail" element={<DetailPage />} />
+                <Route path="/hotel-detail/:id" element={<DetailPage />} />
```

Comment:
두 Detail라우트가 같은 `element`를 바라보고 있네요
하나만 사용해주세요.
아래와 같이 메인 페이지도 path를 부여하고, detail은 main 페이지의 child로 가는게 맞을 것 같습니다.
```suggestion
<Route path="/main" element={<MainPage />}>
  <Route path=":id" element={<DetailPage />} />
</Route>
```